internal/apis/certmanager/types_issuer.go

```diff
@@ -232,6 +238,28 @@ type VaultAppRole struct {
 	SecretRef cmmeta.SecretKeySelector
 }
 
+// VaultKubernetesAuth is used to authenticate against Vault using a client
+// certificate stored in a Secret.
+type VaultClientCertificateAuth struct {
+	// The Vault mountPath here is the mount path to use when authenticating with
+	// Vault. For example, setting a value to `/v1/auth/foo`, will use the path
+	// `/v1/auth/foo/login` to authenticate with Vault. If unspecified, the
+	// default value "/v1/auth/cert" will be used.
+	// +optional
+	Path string
+
+	// Reference to Kubernetes Secret of type "kubernetes.io/tls" (hence containing
+	// tls.crt and tls.key) used to authenticate to Vault using TLS client
+	// authentication.
+	// +optional
+	SecretName string
+
+	// Name of the certificate role to authenticate against.
+	// If not set, matching any certificate role, if available.
+	// +optional
+	Name string
```

Comment:
I got confused by `name`. If it is the name of a role, why not rely on `spec.vault.role`?

Then I realized it isn't really a "role" in the same sense as the other auth methods,  [TLS Certificates Auth Method](https://developer.hashicorp.com/vault/docs/auth/cert) calls it "cert role". In fact, the path is slightly different from e.g. kubernetes auth:

```bash
/v1/auth/kubernetes/role/foo # Kubernetes auth
/v1/auth/cert/certs/foo      # Cert auth
```

Is that why you chose to go with `name` instead of re-using the existing `role`?

Reply:
> Is that why you chose to go with name instead of re-using the existing role?
> 
> (I also saw that AppRole also has its own roleId... weird)

Exactly. I also struggled to find a good definition of "role" (as opposed to e. g. "certificate role"), so I decided to just mirror the vault API as closely as possible (consistent with the way `VaultKubernetesAuth` and `VaultAppRole` do): Just "exposing" the documented parameters from the Vault API: Which is `role_id` (plus `secret_id`) for [AppRole Login](https://developer.hashicorp.com/vault/api-docs/auth/approle#login-with-approle), `role` for [Kubernetes Login](https://developer.hashicorp.com/vault/api-docs/auth/kubernetes#login) and optional `name` for [TLS Cert Login](https://developer.hashicorp.com/vault/api-docs/auth/cert#login-with-tls-certificate-method).

Reply:
> If it is the name of a role, why not rely on spec.vault.role?

On a second look: I cannot find `spec.vault.role` in an [`Issuer`](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.Issuer) under [`spec.vault`](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.VaultIssuer). That would also explain why the other login methods use their own `role_id`/`role`.

Reply:
You are totally correct, I think I made up this `spec.vault.role`. Apologies! The field `name` makes sense since it maps 1-to-1 to the Vault terminology.